python/ray/autoscaler/node_provider.py

```diff
@@ -125,8 +125,11 @@ def terminate_node(self, node_id: str) -> None:
         """Terminates the specified node."""
         raise NotImplementedError
 
-    def terminate_nodes(self, node_ids: List[str]) -> None:
-        """Terminates a set of nodes. May be overridden with a batch method."""
+    def terminate_nodes(self, node_ids: List[str]) -> Optional[Dict[str, Any]]:
+        """Terminates a set of nodes. May be overridden with a batch method,
+        which optionally may return a mapping from deleted node ids to node
+        metadata.
+        """
```

Comment:
Should we also (optionally) return this mapping from `terminate_node`?

Reply:
ehh...yeah, had better as some node providers define terminate_node(node) as
terminate_nodes([node])